The original File EdStatsData.csv has the data in many differnt groups, such as regions like Arab World, East Asia and Pacific etc, 
income groups like low income, high income etc and according to countries as well.
After careful analysis of the data we realize that its is better to use the group based data and drop the country specific data.
This is because the group based data gives us the relevant correlations between wealth and education while freeing us from the 
country data which hard to manage and vizualize as there are 216 countries and areas.

In [8]:
import pandas as pd

In [9]:
#reading the original file
df = pd.read_csv("EdStatsData.csv") 

In [6]:
df

Country Name Country Code  \
0        Arab World          ARB   
1        Arab World          ARB   
2        Arab World          ARB   
3        Arab World          ARB   
4        Arab World          ARB   
5        Arab World          ARB   
6        Arab World          ARB   
7        Arab World          ARB   
8        Arab World          ARB   
9        Arab World          ARB   
10       Arab World          ARB   
11       Arab World          ARB   
12       Arab World          ARB   
13       Arab World          ARB   
14       Arab World          ARB   
15       Arab World          ARB   
16       Arab World          ARB   
17       Arab World          ARB   
18       Arab World          ARB   
19       Arab World          ARB   
20       Arab World          ARB   
21       Arab World          ARB   
22       Arab World          ARB   
23       Arab World          ARB   
24       Arab World          ARB   
25       Arab World          ARB   
26       Arab World          ARB   
27       Arab World          ARB   
28       Arab World          ARB   
29       Arab World          ARB   
...             ...          ...   
886900     Zimbabwe          ZWE   
886901     Zimbabwe          ZWE   
886902     Zimbabwe          ZWE   
886903     Zimbabwe          ZWE   
886904     Zimbabwe          ZWE   
886905     Zimbabwe          ZWE   
886906     Zimbabwe          ZWE   
886907     Zimbabwe          ZWE   
886908     Zimbabwe          ZWE   
886909     Zimbabwe          ZWE   
886910     Zimbabwe          ZWE   
886911     Zimbabwe          ZWE   
886912     Zimbabwe          ZWE   
886913     Zimbabwe          ZWE   
886914     Zimbabwe          ZWE   
886915     Zimbabwe          ZWE   
886916     Zimbabwe          ZWE   
886917     Zimbabwe          ZWE   
886918     Zimbabwe          ZWE   
886919     Zimbabwe          ZWE   
886920     Zimbabwe          ZWE   
886921     Zimbabwe          ZWE   
886922     Zimbabwe          ZWE   
886923     Zimbabwe          ZWE   
886924     Zimbabwe          ZWE   
886925     Zimbabwe          ZWE   
886926     Zimbabwe          ZWE   
886927     Zimbabwe          ZWE   
886928     Zimbabwe          ZWE   
886929     Zimbabwe          ZWE   

                                           Indicator Name  \
0       Adjusted net enrolment rate, lower secondary, ...   
1       Adjusted net enrolment rate, lower secondary, ...   
2       Adjusted net enrolment rate, lower secondary, ...   
3       Adjusted net enrolment rate, lower secondary, ...   
4       Adjusted net enrolment rate, primary, both sex...   
5        Adjusted net enrolment rate, primary, female (%)   
6       Adjusted net enrolment rate, primary, gender p...   
7          Adjusted net enrolment rate, primary, male (%)   
8       Adjusted net enrolment rate, upper secondary, ...   
9       Adjusted net enrolment rate, upper secondary, ...   
10      Adjusted net enrolment rate, upper secondary, ...   
11      Adjusted net enrolment rate, upper secondary, ...   
12      Adjusted net intake rate to Grade 1 of primary...   
13      Adjusted net intake rate to Grade 1 of primary...   
14      Adjusted net intake rate to Grade 1 of primary...   
15      Adjusted net intake rate to Grade 1 of primary...   
16       Adult illiterate population, 15+ years, % female   
17      Adult illiterate population, 15+ years, both s...   
18      Adult illiterate population, 15+ years, female...   
19      Adult illiterate population, 15+ years, male (...   
20      Adult literacy rate, population 15+ years, bot...   
21      Adult literacy rate, population 15+ years, fem...   
22      Adult literacy rate, population 15+ years, gen...   
23      Adult literacy rate, population 15+ years, mal...   
24      Africa Dataset: Average number of grades per m...   
25      Africa Dataset: Average number of pupils per m...   
26      Africa Dataset: Average number of pupils per m...   
27      Africa Dataset: Average number of pupils per m...   
28   

In [10]:
print(len(df["Indicator Name"]))
print(type(df["Indicator Name"]))
print(type(df["Indicator Name"][3]))

886930
<class 'pandas.core.series.Series'>
<class 'str'>


In [11]:
new_df = df.loc[:91624]
new_df.to_csv('primary.csv')

In [13]:
print(len(df["Indicator Name"]) - len(new_df["Indicator Name"]))

795305


In [14]:
print((len(df["Indicator Name"]) - len(new_df["Indicator Name"]))/len(df["Indicator Name"]))

0.8966942148760331


As we see above, we reduce the file size by nearly 89.7% while still keeping the relevant data